network/src/connectivity_manager/mod.rs

```diff
@@ -211,29 +216,61 @@ where
 
         // When we first startup, let's attempt to connect with our seed peers.
         self.check_connectivity(&mut pending_dials).await;
-
+        send_struct_log!(
+            StructuredLogEntry::new_named(logging::CONNECTIVITY_MANAGER_LOOP)
+                .data(logging::NETWORK_CONTEXT, &self.network_context)
+                .data(logging::TYPE, logging::START)
+        );
         trace!("{} Starting connection manager", self.network_context);
         loop {
             self.event_id = self.event_id.wrapping_add(1);
             ::futures::select! {
                 _ = self.ticker.select_next_some() => {
-                    trace!("{} Event Id: {}, type: Tick", self.network_context, self.event_id);
+                    send_struct_log!(StructuredLogEntry::new_named(logging::CONNECTIVITY_MANAGER_LOOP)
+                        .data(logging::NETWORK_CONTEXT, &self.network_context)
+                        .data(logging::TYPE, "tick")
+                        .data(logging::EVENT_ID, self.event_id)
+                    );
+                    trace!("{} Event Id: {}, type: Tick", &self.network_context, self.event_id);
                     self.check_connectivity(&mut pending_dials).await;
                 },
                 req = self.requests_rx.select_next_some() => {
+                    send_struct_log!(StructuredLogEntry::new_named(logging::CONNECTIVITY_MANAGER_LOOP)
+                        .data(logging::NETWORK_CONTEXT, &self.network_context)
+                        .data(logging::TYPE, "connectivity_request")
+                        .data(logging::EVENT_ID, self.event_id)
+                        .data(logging::REQUEST, &req)
+                    );
                     trace!("{} Event Id: {}, type: ConnectivityRequest, req: {:?}", self.network_context, self.event_id, req);
                     self.handle_request(req);
                 },
                 notif = self.connection_notifs_rx.select_next_some() => {
+                    send_struct_log!(StructuredLogEntry::new_named(logging::CONNECTIVITY_MANAGER_LOOP)
```

Comment:
@andll, I feel we could do better here... I know why we ended up with this type of macro, but I think something like:
```
struct_log!(logging::CONNECTIVITY_MANAGER_LOOP)
    .data(...)
    .log();
```
would be a little nicer.

Reply:
What is `CONNECTIVITY_MANAGER_LOOP` here? Your example is pretty much how it is right now, no?

Reply:
It's a little different, I think what he's going for is removing the constructor part.
```
 send_struct_log!(StructuredLogEntry::new_named(logging::CONNECTIVITY_MANAGER_LOOP)
  .data(...)
```
to using the macro to create it, similar to the `debug!` or `info!` macros

```
 struct_log!(logging::CONNECTIVITY_MANAGER_LOOP)
  .data(...)
```

Reply:
Yeah, we have a wrapper around the object so that we can add additional fields. My suggestion is to just make the macro contain both the constructor and additional annotation.

Reply:
Oh. removing constructor will also remove type hints support from IDE for those who use it(like me)

Reply:
FYI, my IDE really does not like the macro, and the type hinting / autoformatting doesn't work within the macro.  I end up making it outside, and then sticking it in the macro.

Reply:
So there are two separate proposals
- wrapper object with specific fields. I am not sure it needs any support in logger crate, I just imaging is as declaring some `struct NetworkLogEntry(StructuredLogEntry)` with appropriate methods?
- Changing macro so that it's shorter, e.g. `struct_log!(logging::CONNECTIVITY_MANAGER_LOOP` instead of `struct_log!(logging::CONNECTIVITY_MANAGER_LOOP::new(...)`. I think this will save maybe 3-4 characters on `::new(`, but will also remove auto-complete support from IDE. is it worth it?

Reply:
I think we should move this discussion to an issue: https://github.com/libra/libra/issues/4516